src/storage/src/source/kafka.rs

```diff
@@ -1581,10 +1480,243 @@ fn fetch_partition_info<C: ConsumerContext>(
     Ok(result)
 }
 
+/// An update produced by the metadata fetcher.
+///
+/// Either the IDs and high watermarks of the topic partitions as of `timestamp`, or a health
+/// status describing a fetch error.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+struct MetadataUpdate {
+    timestamp: mz_repr::Timestamp,
+    info: Result<BTreeMap<PartitionId, HighWatermark>, HealthStatus>,
+}
+
+impl MetadataUpdate {
+    fn to_probe(&self) -> Option<Probe<KafkaTimestamp>> {
```

Comment:
Interesting, was storing an `Antichain` problematic with moving it over Exchange edges?

Reply:
Sorry, I don't follow! This returns an option because a metadata update can only be converted into a probe if it reports a successful metadata fetch (i.e. the `info` is `Ok`, not `Err`). How would you suggest to use an `Antichain` instead?

Reply:
Oh, I was wondering why the type is not `Result<Antichain<KafkaTimestamp>, ..>`. The map from partition id to offset is some kind of frontier

Reply:
I looked at this a bit and passing an `Antichain` seems annoying because the reader operator receiving these updates wants to get at the partition IDs and the high watermarks, so we'd have to convert the `Antichain` back into the original map. At which point it's easier to just pass the map directly. Unless I'm missing a motivation for wanting to use an `Antichain`?

Reply:
I've found it good hygiene to use an `Antichain` in places that represent a frontier but I also see in this particular case we want to pick out the individual partition on the other end so we can delay the conversion.